tools/tests/com.liferay.ide.project.core.tests/src/com/liferay/ide/project/core/tests/modules/jsf/NewLiferayJSFModuleProjectOpTests.java

```diff
@@ -0,0 +1,209 @@
+        Status projectNameExistedValidationStatus = newProjectNameop.getProjectName().validation();
+        assertEquals(
+            projectNameExistedValidationStatus.message(), "A project with that name(ignore case) already exists." );
+    }
+
+    @Test
+    public void testNewLiferayJSFModuleProjectOpBuildType() throws Exception
+    {
+        String[] buildTypesArray = { "gradle-jsf-module", "maven-jsf-module" };
+        NewLiferayJSFModuleProjectOp op = NewLiferayJSFModuleProjectOp.TYPE.instantiate();
+        PossibleValuesService buildTypeValuesService = op.getProjectProvider().service( PossibleValuesService.class );
+        Set<String> buildTypes = buildTypeValuesService.values();
+
+        ArrayList<String> buildTypesList = new ArrayList<String>();
+        for( String buildType : buildTypes )
+        {
+            buildTypesList.add( buildType );
+        }
+
+        String[] typesArray = buildTypesList.toArray( new String[2] );
+
+        assertArrayEquals( buildTypesArray, typesArray );
+    }
+
+    @Test
+    public void testNewLiferayJSFModuleProjectOpDefaultBuildType() throws Exception
+    {
+        NewLiferayJSFModuleProjectOp existedProjectop = NewLiferayJSFModuleProjectOp.TYPE.instantiate();
+        existedProjectop.setProjectName( "Test3" );
+        existedProjectop.setProjectProvider( "maven-jsf-module" );
+        Status exStatus = NewLiferayJSFModuleProjectOpMethods.execute(
+            existedProjectop, ProgressMonitorBridge.create( new NullProgressMonitor() ) );
+        assertTrue( exStatus.ok() );
+
+        NewLiferayJSFModuleProjectOp newBuildTypeop = NewLiferayJSFModuleProjectOp.TYPE.instantiate();
+        DefaultValueService buildTypeDefaultService =
+            newBuildTypeop.getProjectProvider().service( DefaultValueService.class );
+        assertEquals( buildTypeDefaultService.value(), "maven-jsf-module" );
+    }
+
+    @Test
+    public void testNewLiferayJSFModuleProjectOpJsfTemplate() throws Exception
+    {
+        String[] jsfTemplateArray = { "JSF Standard", "Liferay Faces Alloy", "ICEFaces", "PrimeFaces", "RichFaces" };
+        NewLiferayJSFModuleProjectOp op = NewLiferayJSFModuleProjectOp.TYPE.instantiate();
+        PossibleValuesService jsfTemplateValuesService = op.getJsfTemplateName().service( PossibleValuesService.class );
+        Set<String> jsfTemplates = jsfTemplateValuesService.values();
+
+        ArrayList<String> jsfTemplatesList = new ArrayList<String>();
+        for( String jsfTemplate : jsfTemplates )
+        {
+            jsfTemplatesList.add( jsfTemplate );
+        }
+
+        String[] templateArray = jsfTemplatesList.toArray( new String[2] );
+
+        assertArrayEquals( templateArray, jsfTemplateArray );
+    }
+
+    @Test
+    public void testNewLiferayJSFModuleStandardAloneProject() throws Exception
+    {
+        NewLiferayJSFModuleProjectOp mavenProjectop = NewLiferayJSFModuleProjectOp.TYPE.instantiate();
```

Comment:
Need to build the projects to assert there are no problems with the build (no markers)